### PR TITLE
Structure overhaul

### PR DESCRIPTION
**Pull request by patrys**
- Split documentation into several pages with separate sidebars
- Unified the structure somewhat
- Removed what looks like accidental commits to the API reference
- Got rid of some complex custom sidebar formatting
- Dropped giscus and "was this page helpful?"

### CODE REVIEW

File: src/css/components/menu.css [hunk 1]
```diff
@@ -15,28 +15,10 @@
 
 .menu__link {
   font-size: var(--font-size-sm);
-  color: var(--gray10);
   transition: all 0.3s ease-in-out;
```
Comment (peelar) [inline src/css/components/menu.css:18]: **suggestion**:
(I hope this is the right place)

IMO, we should decrease the duration of the transition. Moving cursor from top of the sidebar item list to the bottom leaves a background color "smudge" for too long (to my taste).


https://github.com/saleor/saleor-docs/assets/44495184/6cf4319b-57a1-4b97-9a46-6c6b745785ba

**Review by peelar — COMMENTED** [review]
- "Core Concepts" is missing an overview page
image

- "Core Concepts" has its item in both the sidebar navigation and top navigation
Zrzut ekranu 2024-06-19 o 12 42 50

File: src/css/components/menu.css (hunk 1 above)
Comment (patrys) [reply src/css/components/menu.css:18]: This is completely unrelated to this PR, though. I only reverted even uglier sidebar styling to its original state.

Comment (timuric) [reply src/css/components/menu.css:18]: Current docs don't have group pages, so we would need to solve it somehow

**Comment by patrys** [comment]
@peelar 

>"Core Concepts" is missing an overview page

You are welcome to write something there. I wanted to have a predictable link target but I don't have the time to create new content.

>"Core Concepts" has its item in both the sidebar navigation and top navigation

That's intentional. The same is true for all the other links from the top nav as they are merely shortcuts.

**Comment by timuric** [comment]
I think "Extending Saleor" belongs to core concept

File: src/css/components/menu.css (hunk 1 above)
Comment (peelar) [reply src/css/components/menu.css:18]: I disagree it's unrelated. You changed the way the sidebar looks and I was just sharing my opinion that this particular bit, to me, now looks off.

**Comment by timuric** [comment]
About Saleor can not be collapsed as Why saleor
image

File: src/css/components/menu.css (hunk 1 above)
Comment (patrys) [reply src/css/components/menu.css]: I removed all easing from the sidebar, so the animation is gone entirely.

**Comment by patrys** [comment]
@timuric

>I think "Extending Saleor" belongs to core concept

I wanted "core concepts" to be about using Saleor, not building stuff around it. You are welcome to propose a better name for the section. Medusa calls a similar thing "Commerce Modules."

**Review by maarcingebala — APPROVED** [review]

**Review by aniav — APPROVED** [review]

**Comment by timuric** [comment]
@aniav we can merge now?

**Comment by aniav** [comment]
@timuric had to do path tweaks, once vercel builds I will merge to not block anything any more 🙌

**Pull request merged**